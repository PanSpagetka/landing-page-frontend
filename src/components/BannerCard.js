import React from 'react';
import propTypes from 'prop-types';

import {
    Card, CardBody,
    Flex, FlexItem, FlexModifiers,
    Title
} from '@patternfly/react-core';

import './BannerCard.scss';

const BannerCard = ({ image, title, description, ...props }) => {
    return (
        <Card className='ins-c-banner-card' { ...props }>
            <CardBody>
                <Flex className='pf-m-align-items-center' breakpointMods={ [{ modifier: FlexModifiers.column }] }>
                    <FlexItem className='ins-c-banner-card__icon'>
                        <img
                            aria-hidden
                            src={ image }
                            alt={ `${title} logo` } />
                    </FlexItem>
                    <FlexItem className='ins-c-banner-card__title'><Title headingLevel='h4' size='xl'>{ title }</Title></FlexItem>
                    <FlexItem className='ins-c-banner-card__description'> { description } </FlexItem>
                </Flex>
            </CardBody>
        </Card>
    );
};

export default BannerCard;

BannerCard.propTypes = {
    image: propTypes.any,
    title: propTypes.string,
    description: propTypes.string
};
