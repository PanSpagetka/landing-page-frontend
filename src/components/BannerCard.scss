.ins-c-banner-card {
    text-align: center;
    border-radius: 2px;
    height: 100%;
    .pf-c-card__body {
        padding: var(--pf-global--spacer--xl) var(--pf-global--spacer--md) var(--pf-global--spacer--3xl);
    }
    &__icon {
        margin-bottom: 4px;
        img { height: 100px; }
    }
    &__description {
        color: var(--pf-global--Color--200);
    }
}