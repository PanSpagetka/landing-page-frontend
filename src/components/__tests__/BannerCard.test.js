import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BannerCard from '../BannerCard';

describe('BannerCard component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<BannerCard image='' title='test' description='description'/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
