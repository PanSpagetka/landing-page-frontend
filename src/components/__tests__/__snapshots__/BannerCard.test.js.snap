// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCard component should render correctly 1`] = `
<Card
  className="ins-c-banner-card"
>
  <CardBody>
    <Flex
      breakpointMods={
        Array [
          Object {
            "modifier": "column",
          },
        ]
      }
      className="pf-m-align-items-center"
    >
      <FlexItem
        className="ins-c-banner-card__icon"
      >
        <img
          alt="test logo"
          aria-hidden={true}
          src=""
        />
      </FlexItem>
      <FlexItem
        className="ins-c-banner-card__title"
      >
        <Title
          headingLevel="h4"
          size="xl"
        >
          test
        </Title>
      </FlexItem>
      <FlexItem
        className="ins-c-banner-card__description"
      >
         
        description
         
      </FlexItem>
    </Flex>
  </CardBody>
</Card>
`;
